API/fleece/slice.hh

```diff
@@ -238,7 +238,8 @@ namespace fleece {
 
         void setBuf(const void *b NONNULL) noexcept          {pure_slice::setBuf(b);}
         void setSize(size_t s) noexcept                      {pure_slice::setSize(s);}
-        void shorten(size_t s)                      {assert_precondition(s <= size); setSize(s);}
+        void shorten(size_t s)                               {assert_precondition(s <= size); setSize(s);}
+        void clear()                                         { memset((void *)buf, 0, size); }
```

Comment:
The name `clear` doesn't seem right to me. STL containers have `clear` methods that reset them to zero size, and LiteCore has some methods like that too. How about `wipe` or `zeroFill`?